internal/server/auth/middleware/grpc/middleware.go

```diff
@@ -196,9 +205,32 @@ func JWTAuthenticationInterceptor(logger *zap.Logger, validator jwt.Validator, e
 			return ctx, ErrUnauthenticated
 		}
 
+		metadata := map[string]string{}
+
+		userClaimsRaw, ok := jwtClaims["user"]
+		if ok {
+			userClaimsJSON, err := json.Marshal(userClaimsRaw)
+			if err != nil {
+				return ctx, ErrUnauthenticated
+			}
+
+			userClaims := UserClaims{}
+
+			if err := json.Unmarshal(userClaimsJSON, &userClaims); err != nil {
+				return ctx, ErrUnauthenticated
+			}
+
+			metadata = map[string]string{
+				"io.flipt.auth.jwt.email":   userClaims.Email,
+				"io.flipt.auth.jwt.sub":     userClaims.Sub,
+				"io.flipt.auth.jwt.picture": userClaims.Image,
+				"io.flipt.auth.jwt.name":    userClaims.Name,
+			}
```

Comment:
Take it or leave it: Seems a waste to marshal it out to json and back again just to read some fields. Could do something like:
```suggestion
			metadata = map[string]string{}
			setIfPresent := func(metaField, claimField string) {
				if v, ok := userClaimsRaw[claimField]; ok {
					// coerce into a string
					metadata[metaField] = fmt.Sprintf("%v", v)
				}
			}
			setIfPresent("io.flipt.auth.jwt.email", "email")
			setIfPresent("io.flipt.auth.jwt.sub", "sub")
			setIfPresent("io.flipt.auth.jwt.picture", "image")			
			setIfPresent("io.flipt.auth.jwt.name", "name")
```

Reply:
@GeorgeMac Definitely could have thought of this a bit more, thanks!